packages/twenty-front/src/modules/settings/hooks/useSettingsNavigationItems.tsx

```diff
@@ -0,0 +1,193 @@
+import {
+  IconApps,
+  IconAt,
+  IconCalendarEvent,
+  IconCode,
+  IconColorSwatch,
+  IconComponent,
+  IconCurrencyDollar,
+  IconFlask,
+  IconFunction,
+  IconHierarchy2,
+  IconKey,
+  IconLock,
+  IconMail,
+  IconRocket,
+  IconServer,
+  IconSettings,
+  IconUserCircle,
+  IconUsers,
+} from 'twenty-ui';
+
+import { SettingsPath } from '@/types/SettingsPath';
+import { SettingsFeatures } from 'twenty-shared';
+import { FeatureFlagKey } from '~/generated-metadata/graphql';
+
+import { currentUserState } from '@/auth/states/currentUserState';
+import { billingState } from '@/client-config/states/billingState';
+import { labPublicFeatureFlagsState } from '@/client-config/states/labPublicFeatureFlagsState';
+import { useSettingsPermissionMap } from '@/settings/roles/hooks/useSettingsPermissionMap';
+import { NavigationDrawerItemIndentationLevel } from '@/ui/navigation/navigation-drawer/components/NavigationDrawerItem';
+import { useFeatureFlagsMap } from '@/workspace/hooks/useFeatureFlagsMap';
+import { t } from '@lingui/core/macro';
+import { useRecoilValue } from 'recoil';
+
+export type SettingsNavigationSection = {
+  label: string;
+  items: SettingsNavigationItem[];
+  isAdvanced?: boolean;
+};
+
+export type SettingsNavigationItem = {
+  label: string;
+  path: SettingsPath;
+  Icon: IconComponent;
+  indentationLevel?: NavigationDrawerItemIndentationLevel;
+  matchSubPages?: boolean;
+  isHidden?: boolean;
+  subItems?: SettingsNavigationItem[];
+  isAdvanced?: boolean;
+};
+
+export const useSettingsNavigationItems = (): SettingsNavigationSection[] => {
```

Comment:
Way better thanks ! Also easier for maintaining granular changes about permissions and feature flags